### PR TITLE
Feature/general fixes

### PR DESCRIPTION
**Pull request by ddawson235**
#### What changes are included in this PR?
* Fixed the typescript type casting issue that was preventing build
* Added a CI workflow which includes typescript build checks and prettier style checks (will add eslint later, too many issues for now)
* Style linted all relevant files in the `src/` dir
* Added an `npm run style` (prettier) and `npm run lint` (eslint) that can be executed to make development easier

### CODE REVIEW

**Comment by ddawson235** [comment]
fyi @mmanou @dieffrei a build error exists in the project, have fixed it and added some CI capabilities into the project so they can be aught easier before merging. Once this branch is merged the new version will be published to npm and the SFDX CLI plugin update will be available

**Comment by dieffrei** [comment]
Thanks!

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Checkout repository
+      uses: actions/checkout@master
+    - name: Set up Node.js
+      uses: actions/setup-node@master
+      with:
+        node-version: '10.x'
+    - name: Install packages
+      run: npm i
+    - name: Run TypeScript Build
+      run: tsc --build
```
Comment (patelsu) [inline .github/workflows/check.yml:35]: add new line at the end according to standards

Comment (patelsu) [reply .github/workflows/check.yml:35]: do it on all files that shows red dot at end

**Pull request merged**